From 491b74609ba54bb4dcda6f9b18763a6bcbcf5261 Mon Sep 17 00:00:00 2001
From: Peleg Vigodny <peleg.vigodny@tandemg.com>
Date: Tue, 10 Aug 2021 18:02:38 +0300
Subject: [PATCH] [SW-48529] makefiles: use root dir definition

Define and use variable PREBOOT_ROOT_DIR, which reflects the root
directory from which the preboot is built.

This variable will be defined in the primary makefiles and used there
internally in order to facilitate future changes in the preboot
directory hierarchy.

Furthermore, it will be used in internal architecture-specific Makefiles
when referring to files inside the preboot build directory, allowing
them to be included by the primary makefiles.

Change-Id: Ib5af127b7a333de7145fdee11a87c2bcbc47eeb4
Signed-off-by: Peleg Vigodny <peleg.vigodny@tandemg.com>
---

diff --git a/Makefile b/Makefile
index 531ddeb..2c1c9c4 100644
--- a/Makefile
+++ b/Makefile
@@ -6,6 +6,7 @@
 ifeq ($(HL_DEV_SUBTYPE),)
 	HL_DEV_SUBTYPE := arm
 endif
+PREBOOT_ROOT_DIR := .
 HL_DEV_ARCH := arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
 HL_DEV_ARCH_COMMON := arch/$(BASE_DEV_NAME)/common
 ifeq ($(BASE_DEV_NAME),greco)
diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index 7b9c38b..eb6f897 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -2,13 +2,14 @@
 OBJCOPY = $(CROSS_COMPILE)objcopy
 BASE_DEV_NAME := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 HL_DEV_SUBTYPE := $(shell echo $(HL_DEV) | cut -d'_' -s -f2,3)
-HL_DEV_ARCH := arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
+PREBOOT_ROOT_DIR := ../..
+HL_DEV_ARCH := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
 
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
-CFLAGS += -I ./../../include/ -I ./../../ -I ./../../lib/common/ -I ./../../$(HL_DEV_ARCH)/
+CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(HL_DEV_ARCH)/
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
@@ -22,14 +23,14 @@
 EROM_APP_OBJECTS := $(EROM_APP_S:.S=.o)
 
 EROM_PCIE_DEPS := \
-	../../lib/common/pll.o \
-	../../lib/common/pci.o \
-	../../lib/common/uart.o \
-	../../app/ppboot/ppboot.o \
-	../../$(HL_DEV_ARCH)/$(HL_TARGET).o \
-	../../$(HL_DEV_ARCH)/../common/$(HL_TARGET).o \
-	../../$(HL_DEV_ARCH)/../common/common.o \
-	../../$(HL_DEV_ARCH)/common.o
+	$(PREBOOT_ROOT_DIR)/lib/common/pll.o \
+	$(PREBOOT_ROOT_DIR)/lib/common/pci.o \
+	$(PREBOOT_ROOT_DIR)/lib/common/uart.o \
+	$(PREBOOT_ROOT_DIR)/app/ppboot/ppboot.o \
+	$(HL_DEV_ARCH)/$(HL_TARGET).o \
+	$(HL_DEV_ARCH)/../common/$(HL_TARGET).o \
+	$(HL_DEV_ARCH)/../common/common.o \
+	$(HL_DEV_ARCH)/common.o
 
 .PHONY: all
 all: erom_recovery erom_pcie erom_tpm
diff --git a/app/recovery/Makefile b/app/recovery/Makefile
index 35c433a..a06936e 100644
--- a/app/recovery/Makefile
+++ b/app/recovery/Makefile
@@ -4,6 +4,7 @@
 HL_DEV_SUBTYPE := $(shell echo $(HL_DEV) | cut -d'_' -s -f2,3)
 HL_DEV_ARCH := arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
 HL_DEV_ARCH_COMMON := arch/$(BASE_DEV_NAME)/common
+PREBOOT_ROOT_DIR := ../..
 ifeq ($(BASE_DEV_NAME),greco)
 	BASE_DEV_NAME := goya2
 endif
@@ -13,22 +14,22 @@
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
-CFLAGS += -I ./../../include/ -I ./../../ -I ./../../lib/common/ -I ./../../$(HL_DEV_ARCH)/
+CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
-RECOVERY_LD_SCRIPT := ./../../$(HL_DEV_ARCH)/recovery.lds
+RECOVERY_LD_SCRIPT := $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/recovery.lds
 LDFLAGS := -g -nostdlib -static
 LDFLAGS += -Wl,--gc-sections
 LDLIBS := -lgcc
 
 RECOVERY_SRCS_C := $(shell ls *.c)
-RECOVERY_SRCS_C += $(shell ls ../../lib/common/*.c)
-RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH_COMMON)/common.c)
-RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/common.c)
-RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/$(HL_TARGET).c)
-RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH_COMMON)/$(HL_TARGET).c)
+RECOVERY_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/common/*.c)
+RECOVERY_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/common.c)
+RECOVERY_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/common.c)
+RECOVERY_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/$(HL_TARGET).c)
+RECOVERY_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/$(HL_TARGET).c)
 RECOVERY_APP_S := $(shell ls crt0_recovery.S)
 
 RECOVERY_OBJECTS := $(RECOVERY_SRCS_C:.c=.o)
diff --git a/arch/gaudi2/arc/Makefile b/arch/gaudi2/arc/Makefile
index c58b8bf..c62ae85 100644
--- a/arch/gaudi2/arc/Makefile
+++ b/arch/gaudi2/arc/Makefile
@@ -3,8 +3,8 @@
 FW_SIZE := 0x8000
 PPBOOT_MAX_SIZE := 16384
 
-LDFLAGS += -Wl,--just-symbols ../arcboot-custom/exports.elf
+LDFLAGS += -Wl,--just-symbols $(PREBOOT_ROOT_DIR)/../arcboot-custom/exports.elf
 
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/pll.c)
-PREBOOT_SRCS_C += $(shell ls lib/preboot/thermal_v2.c)
+PREBOOT_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/preboot/thermal_v2.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/security.c)
diff --git a/arch/greco/arc/Makefile b/arch/greco/arc/Makefile
index 85d57dd..3f8c5b6 100644
--- a/arch/greco/arc/Makefile
+++ b/arch/greco/arc/Makefile
@@ -3,5 +3,5 @@
 FW_SIZE := 0x8000
 
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/pll.c)
-PREBOOT_SRCS_C += $(shell ls lib/preboot/thermal_v2.c)
+PREBOOT_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/preboot/thermal_v2.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/security.c)
diff --git a/arch/greco/arm/Makefile b/arch/greco/arm/Makefile
index 36a396c..d02991a 100644
--- a/arch/greco/arm/Makefile
+++ b/arch/greco/arm/Makefile
@@ -2,7 +2,7 @@
 
 FW_SIZE := 0x4400
 PPBOOT_MAX_SIZE := 18432
-PREBOOT_SRCS_C += $(shell ls lib/preboot/thermal_v2.c)
+PREBOOT_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/preboot/thermal_v2.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV_ARCH)/security.c)
 
 CFLAGS += -mstrict-align
